### ISSUE TITLE
Homepage X-Robots-Tag: none. 

### ISSUE THREAD

**Issue by adcalex** [putyourlightson/craft-blitz#511]
Please provide a clear and concise description of what the bug is, including:
- Steps to reproduce the issue.

**After uninstall blitz plugin, all index page for X-Robots-Tag: all. Reinstall blitz, only homepage has X-Robots-Tag:none and all other pages were showing X-Robots-Tag: all** 

- A clear and concise description of what you expected to happen.
**We expect all indexed pages to have X-Robots-Tag: all**

- If applicable, add screenshots to help explain your problem.
https://github.com/putyourlightson/craft-blitz/blob/fa758238337818cd87669fd09cdd14c116cf9bb7/src/services/CacheRequestService.php#L542

**We have already added the disallowRobots: False in the general config.** 

<img width="424" alt="Screen Shot 2023-05-30 at 9 57 26 AM" src="https://github.com/putyourlightson/craft-blitz/assets/51177864/f950e844-97d9-4de9-94be-319fe738a8f3">

- The plugin and Craft version numbers.

**Application Info
PHP version 7.4.16
OS version  Linux 3.10.0-1160.76.1.el7.x86_64
Database driver & version   MySQL 8.0.30
Image driver & version  Imagick 3.4.4 (ImageMagick 6.9.12-55)
Craft edition & version Craft Pro 3.7.61
Yii version 2.0.45
Twig version    v2.15.4
Guzzle version  7.5.0
Plugins
Agent   4.0.0
Amazon S3   1.3.0
Asset Rev   6.0.2
Blitz   3.12.7
Classnames  1.0.3
Feed Me 4.5.3
Field Manager   2.2.5
GeoMate v1.3.1.2
Imgix   2.1.0
KeyChain    1.0.3
MatrixMate  1.5.0
Neo 2.13.16
Redactor    2.10.10
Redactor Custom Styles  3.0.4
Retour  3.2.10
SAML Service Provider   2.7.5
SEOmatic    3.4.53
Super Table 2.7.3
Tags    1.0.9**
Twig Perversion 2.2.0

**Comment by bencroker** [comment]
> We expect all indexed pages to have X-Robots-Tag: all

> We have already added the disallowRobots: True in the general config.

If you want pages to have `X-Robots-Tag: all` then please set the `disallowRobots` setting to `false`.

**Comment by adcalex** [comment]
**Double checked the env file and the general.php config file.  it is on false. Sorry about the confusion.** 
cat .env | grep -i disall
DISALLOW_ROBOTS=false

cat general.php | grep -i disallow
        'disallowRobots' => false,
        'disallowRobots' => false,
        'disallowRobots' => getenv('DISALLOW_ROBOTS'),

**Looks like the homepage still on "None" for x-robots-tag.** 


**Additionally, here is my general.php looks like.** 

`<?php
/**
 * General Configuration
 *
 * All of your system's general configuration settings go in here. You can see a
 * list of the available settings in vendor/craftcms/cms/src/config/GeneralConfig.php.
 *
 * @see \craft\config\GeneralConfig
 */

return [
    // Global settings
    '*' => [
        'addTrailingSlashesToUrls' => false,
        'aliases' => [
            '@web' => getenv('DEFAULT_SITE_URL'),
            '@webroot' => dirname(__DIR__) . '/web',
            '@storage' => getenv('STORAGE_PATH'),
        ],
        'allowAdminChanges' => false,
        'autosaveDrafts'    => false,
        'cpTrigger' => getenv('CP_TRIGGER'),
        'defaultWeekStartDay' => 1,
        'enableCsrfProtection' => true,
        'enableGql' => true,
        'enableTemplateCaching' => true,
        'extraAllowedFileExtensions' => 'eps',
        'loginPath' => getenv('CP_TRIGGER') . '/login',
        'disallowRobots' => false,
        'backupOnUpdate' => false,
    ],

    // Dev environment settings
    'dev' => [
        // Dev Mode (see https://craftcms.com/guides/what-dev-mode-does)
        'allowAdminChanges' => true,
        'devMode' => true,
        'disallowRobots' => false,
        // Local database backups fix for MAMP users
        'backupCommand' => getenv('BACKUP_COMMAND'),
        'restoreCommand' => getenv('RESTORE_COMMAND'),
        'enableTemplateCaching' => false,
    ],

    'production' => [
        'disallowRobots' => getenv('DISALLOW_ROBOTS'),
    ],
];`

**Comment by bencroker** [comment]
My guess is that the homepage is still cached from before you made your changes. Can you trying clearing, and if relevant purging, the cache for the homepage and then retesting?

**Comment by adc-tgupta** [comment]
@bencroker when we set disallowRobots setting to false, X-Robots-Tag is not set for homepage. all other pages seem to be fine.  is there any other setting you would like us to try.

**Comment by bencroker** [comment]
Can you try setting the value to `DISALLOW_ROBOTS=0` in your `.env` file? Environment variables are always strings, so I’m not sure if using `false` is possible here.

**Comment by adc-tgupta** [comment]
@bencroker  I had tried both ways, by using directly  'disallowRobots' => false, and by using 'disallowRobots' => getenv('DISALLOW_ROBOTS'), both give same result.

**Comment by bencroker** [comment]
Can you send a copy of the database, along with the composer files and general config file, to support@putyourlightson.com and I’ll take a closer look?

**Comment by bencroker** [comment]
Closing, as this appears to be unrelated to Blitz.

**Comment by asethii** [comment]
Closing the loop here, in case anyone runs into this, in the future. 

We ended up updating our craft general.php config file to fix this. We had previously set disallowRobots in production like this:

'production' => [
        'disallowRobots' => getenv('DISALLOW_ROBOTS'),
    ],
    
After removing this, we were seeing the expected results. I'm not totally sure how this resolved our issue, but my guess it might be something to do with this line - https://github.com/putyourlightson/craft-blitz/blob/fa758238337818cd87669fd09cdd14c116cf9bb7/src/services/CacheRequestService.php#L539
